src/sonic-framework/tests/rebootbe_test.cpp

```diff
@@ -0,0 +1,420 @@
+  }
+
+  void SendRebootViaProto(RebootRequest &request) {
+    std::string json_string;
+    gpu::MessageToJsonString(request, &json_string);
+
+    SendRebootRequest("Reboot", "StatusCode", DATA_TUPLE_KEY, json_string);
+  }
+
+  void SendRebootStatusRequest(void) {
+    SendRebootRequest("RebootStatus", "StatusCode", DATA_TUPLE_KEY,
+                      "json status request");
+  }
+
+  void start_reboot_via_rpc(
+      RebootRequest &request,
+      swss::StatusCode expected_result = swss::StatusCode::SWSS_RC_SUCCESS) {
+    SendRebootViaProto(request);
+    while (true) {
+      int ret;
+      swss::Selectable *sel;
+      ret = m_s.select(&sel, SELECT_TIMEOUT_250_MS);
+      if (ret != swss::Select::OBJECT) continue;
+      if (sel != &m_rebootbeReponseChannel) continue;
+      break;
+    }
+    std::string op, data;
+    std::vector<swss::FieldValueTuple> ret_values;
+    m_rebootbeReponseChannel.pop(op, data, ret_values);
+
+    EXPECT_THAT(op, StrEq("Reboot"));
+    EXPECT_THAT(data, StrEq(swss::statusCodeToStr(expected_result)));
+  }
+
+  gnoi::system::RebootStatusResponse do_reboot_status_rpc() {
+    SendRebootStatusRequest();
+    while (true) {
+      int ret;
+      swss::Selectable *sel;
+      ret = m_s.select(&sel, SELECT_TIMEOUT_250_MS);
+      if (ret != swss::Select::OBJECT) continue;
+      if (sel != &m_rebootbeReponseChannel) continue;
+      break;
+    }
+    std::string op, data;
+    std::vector<swss::FieldValueTuple> ret_values;
+    m_rebootbeReponseChannel.pop(op, data, ret_values);
+
+    EXPECT_THAT(op, StrEq("RebootStatus"));
+    EXPECT_EQ(data, swss::statusCodeToStr(swss::StatusCode::SWSS_RC_SUCCESS));
+
+    std::string json_response;
+    for (auto &fv : ret_values) {
+      if (DATA_TUPLE_KEY == fvField(fv)) {
+        json_response = fvValue(fv);
+      }
+    }
+    gnoi::system::RebootStatusResponse response;
+    gpu::JsonStringToMessage(json_response, &response);
+    return response;
+  }
+
+  void GetNotificationResponse(swss::NotificationConsumer &consumer,
+                               std::string &op, std::string &data,
+                               std::vector<swss::FieldValueTuple> &values) {
+    swss::Select s;
+    s.addSelectable(&consumer);
+    swss::Selectable *sel;
+    s.select(&sel, SELECT_TIMEOUT_250_MS);
+
+    consumer.pop(op, data, values);
+  }
+
+  NotificationResponse handle_reboot_request(std::string &json_request) {
+    return m_rebootbe.handle_reboot_request(json_request);
+  }
+
+
+  // Mock interfaces.
+  NiceMock<MockDbusInterface> m_dbus_interface;
+
+  // DB connectors
+  swss::DBConnector m_db;
+  swss::DBConnector m_config_db;
+
+  // Reboot thread signaling.
+  swss::NotificationProducer m_rebootbeRequestChannel;
+  swss::Select m_s;
+  swss::NotificationConsumer m_rebootbeReponseChannel;
+
+  // Module under test.
+  std::unique_ptr<std::thread> m_rebootbe_thread;
+  RebootBE m_rebootbe;
+
+};
+
+class RebootBETest : public RebootBETestWithoutStop {
```

Comment:
This is pretty counter-intuitive. From the name RebootBETest seems to be of a broader scope of RebootBETestWithoutStop, but here it becomes a subclass of RebootBETestWithoutStop. Can you either document this or maybe rename the parent class to be RebootBETest and the child class to be RebootBETestWithStop?